type-definitions/most.d.ts

```diff
@@ -0,0 +1,349 @@
+    d: Stream<D>,
+    e: Stream<E>
+  ): Stream<R>;
+
+  recoverWith<B>(p: (a: B) => A): Stream<A>;
+  multicast(): Stream<A>;
+
+  thru<B>(transform: (stream: Stream<A>) => Stream<B>): Stream<B>;
+}
+
+declare interface DisposeFn {
+  (): void|Promise<any>;
+}
+
+export function create<A>(f: (add: (a: A) => any, end: (x: any) => any, error: (e: Error) => any) => void|DisposeFn): Stream<A>;
+export function just<A>(a: A): Stream<A>;
+export function of<A>(a: A): Stream<A>;
+export function empty(): Stream<any>;
+export function never(): Stream<any>;
+export function from<A>(as: Iterable<A>): Stream<A>;
+export function periodic<A>(period: number, a?: A): Stream<A>;
+export function fromEvent(event: string, target: any, useCapture?: boolean): Stream<Event>;
+
+export function unfold<A, B, S>(f: (seed: S) => SeedValue<S, B|Promise<B>>, seed: S): Stream<B>;
+export function iterate<A>(f: (a: A) => A|Promise<A>, a: A): Stream<A>;
+export function generate<A>(g: CreateGenerator<A>, ...args: Array<any>): Stream<A>;
+
+export function reduce<A, B>(f: (b: B, a: A) => B, b: B, s: Stream<A>): Promise<B>;
+export function observe<A>(f: (a: A) => any, s: Stream<A>): Promise<any>;
+export function forEach<A>(f: (a: A) => any, s: Stream<A>): Promise<any>;
+export function drain<A>(s: Stream<A>): Promise<any>;
+
+export function constant<A, B>(b: B, s: Stream<A>): Stream<B>;
+export function map<A, B>(f: (a: A) => B, s: Stream<A>): Stream<B>;
+export function tap<A>(f: (a: A) => any, s: Stream<A>): Stream<A>;
+export function ap<A, B>(fs: Stream<(a: A) => B>, as: Stream<A> ): Stream<B>;
+export function chain<A, B>(f: (a: A) => Stream<B>, s: Stream<A>): Stream<B>;
+export function flatMap<A, B>(f: (a: A) => Stream<B>, s: Stream<A>): Stream<B>;
+export function join<A>(s: Stream<Stream<A>>): Stream<A>;
+export function switchLatest<A>(s: Stream<Stream<A>>): Stream<A>;
+
+export function continueWith<A>(f: (a: any) => Stream<A>, s: Stream<A>): Stream<A>;
+export function concatMap<A, B>(f: (a: A) => Stream<B>, s: Stream<A>): Stream<B>;
+export function mergeConcurrently<A>(concurrency: number, s: Stream<Stream<A>>): Stream<A>;
+
+export function merge<A>(...ss: Array<Stream<A>>): Stream<A>;
+export function mergeArray<A>(streams: Array<Stream<A>>): Stream<A>;
+
+export function combine<A, B, R>(
+  fn: (a: A, b: B) => R,
+  a: Stream<A>,
+  b: Stream<B>
+): Stream<R>;
+export function combine<A, B, C, R>(
+  fn: (a: A, b: B, c: C) => R,
+  a: Stream<A>,
+  b: Stream<B>,
+  c: Stream<C>
+): Stream<R>;
+export function combine<A, B, C, D, R>(
+  fn: (a: A, b: B, c: C, d: D) => R,
+  a: Stream<A>,
+  b: Stream<B>,
+  c: Stream<C>,
+  d: Stream<D>
+): Stream<R>;
+export function combine<A, B, C, D, E, R>(
+  fn: (a: A, b: B, c: C, d: D, e: E) => R,
+  a: Stream<A>,
+  b: Stream<B>,
+  c: Stream<C>,
+  d: Stream<D>,
+  e: Stream<E>
+): Stream<R>;
+
+export function combineArray<A, B, R>(
+  fn: (a: A, b: B) => R,
+  array: [Stream<A>, Stream<B>]
+): Stream<R>;
```

Comment:
I really like that you've enforced the minimum number of array elements (as well as varargs above) to 2. :+1: